src/EditorFeatures/Core/Classification/TotalClassificationTaggerProvider.cs

```diff
@@ -69,26 +69,26 @@ internal sealed class TotalClassificationTaggerProvider(
     }
 
     internal sealed class TotalClassificationAggregateTagger(
-        SyntacticClassificationTaggerProvider.Tagger syntacticTagger,
-        SimpleAggregateTagger<IClassificationTag> semanticTagger,
-        SimpleAggregateTagger<IClassificationTag> embeddedTagger)
-        : AbstractAggregateTagger<IClassificationTag>(ImmutableArray.Create<ITagger<IClassificationTag>>(syntacticTagger, semanticTagger, embeddedTagger))
+        EfficientTagger<IClassificationTag> syntacticTagger,
+        EfficientTagger<IClassificationTag> semanticTagger,
+        EfficientTagger<IClassificationTag> embeddedTagger)
```

Comment:
you will see EfficientTagger at teh end of the review.  Essentially, it is a tagger that can put items into a list, instead of creating a fresh list to pass out each time.  This allows for list pooling.